packages/front-end/components/WatchButton.tsx

```diff
@@ -1,17 +1,30 @@
 import { FC, useState } from "react";
 import { FaEye } from "react-icons/fa";
-import { useAuth } from "../../services/auth";
-import { useWatching } from "../../services/WatchProvider";
+import { useAuth } from "../services/auth";
+import { useWatching } from "../services/WatchProvider";
 
 const WatchButton: FC<{
-  experiment: string;
+  item: string;
+  itemType: "feature" | "experiment";
   type?: "button" | "icon" | "link";
-}> = ({ experiment, type = "button" }) => {
-  const { watching, refreshWatching } = useWatching();
+}> = ({ item, itemType, type = "button" }) => {
+  const {
+    watchedExperiments,
+    watchedFeatures,
+    refreshWatching,
+  } = useWatching();
   const { apiCall } = useAuth();
   const [loading, setLoading] = useState(false);
 
-  const isWatching = watching.includes(experiment);
+  let isWatching;
+  let urlFirst;
+  if (itemType == "feature") {
+    urlFirst = "feature";
+    isWatching = watchedFeatures.includes(item);
+  } else if (itemType == "experiment") {
+    urlFirst = "experiment";
+    isWatching = watchedExperiments.includes(item);
+  }
```

Comment:
Got it! Yeah I only added the urlFirst because before we were worried about attackers hacking the url and all. I changed it back to the dynamic version